dll/cpl/input/settings_page.c

```diff
@@ -11,192 +11,341 @@
+                                          GetSystemMetrics(SM_CXSMICON),
+                                          GetSystemMetrics(SM_CYSMICON),
+                                          0);
+        if (hKeyboardIcon)
+        {
+            KeyboardImageIndex = ImageList_AddIcon(hImageList, hKeyboardIcon);
+            DestroyIcon(hKeyboardIcon);
+        }
 
-    if (hImageList != NULL)
+        // Language
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_TEXT | TVIF_IMAGE | TVIF_PARAM | TVIF_SELECTEDIMAGE | TVIF_STATE;
+        item.pszText        = pInputNode->pLocale->pszName;
+        item.iImage         = LangImageIndex;
+        item.iSelectedImage = LangImageIndex;
+        item.lParam         = LOWORD(pInputNode->pLocale->dwId);
+        if (bBold)
+        {
+            item.state = item.stateMask = TVIS_BOLD;
+        }
+        insert.hParent      = TVI_ROOT;
+        insert.hInsertAfter = TVI_LAST;
+        insert.item         = item;
+        hItem = TreeView_InsertItem(hwndList, &insert);
+
+        // Keyboard
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_TEXT | TVIF_IMAGE | TVIF_PARAM | TVIF_SELECTEDIMAGE | TVIF_STATE;
+        item.pszText        = szKeyboard;
+        item.iImage         = KeyboardImageIndex;
+        item.iSelectedImage = KeyboardImageIndex;
+        item.lParam         = 0xFFFF;
+        if (bBold)
+        {
+            item.state = item.stateMask = TVIS_BOLD;
+        }
+        insert.hParent      = hItem;
+        insert.hInsertAfter = TVI_LAST;
+        insert.item         = item;
+        hItem = TreeView_InsertItem(hwndList, &insert);
+    }
+    else
     {
-        HICON hLayoutIcon;
+        // Language
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_STATE | TVIF_HANDLE;
+        item.hItem          = hItem;
+        item.stateMask      = TVIS_BOLD;
+        if (TreeView_GetItem(hwndList, &item) && bBold && !(item.state & TVIS_BOLD))
+        {
+            item.mask = TVIF_STATE | TVIF_HANDLE;
+            item.hItem = hItem;
+            item.state = item.stateMask = TVIS_BOLD;
+            TreeView_SetItem(hwndList, &item);
+        }
 
-        hLayoutIcon = CreateLayoutIcon(pInputNode->pszIndicator,
-                                       (pInputNode->wFlags & INPUT_LIST_NODE_FLAG_DEFAULT));
-        if (hLayoutIcon != NULL)
+        // Keyboard
+        hItem = TreeView_GetChild(hwndList, hItem);
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_STATE | TVIF_HANDLE;
+        item.hItem          = hItem;
+        item.stateMask      = TVIS_BOLD;
+        if (TreeView_GetItem(hwndList, &item) && bBold && !(item.state & TVIS_BOLD))
         {
-            ImageIndex = ImageList_AddIcon(hImageList, hLayoutIcon);
-            DestroyIcon(hLayoutIcon);
+            item.mask = TVIF_STATE | TVIF_HANDLE;
+            item.hItem = hItem;
```

Comment:
Should both of these attributes (the mask and hItem) be reset yet again?

Reply:
The root "Locale" items can be shared some leaves.
Zero or one leaves can be default.

Reply:
For example:
![ex](https://user-images.githubusercontent.com/2107452/192175820-737a79c1-66d8-42f9-8b52-f04b28f8a58c.png)

Reply:
Win2k3:
![1](https://user-images.githubusercontent.com/2107452/192204144-c2d3929b-b9e9-4e83-96ce-5e2e62b497d1.png)

Windows doesn't bold the keyboard item text...

Reply:
Fixed.